File: backend/Onied/Courses/Dtos/CourseCardDto.cs
namespace Courses.Dtos;

public class CourseCardDto
{
    public int Id { get; set; }
    public string Title { get; set; } = null!;
    public string PictureHref { get; set; } = null!;
    public int Price { get; set; }
    public CategoryDto Category { get; set; } = null!;
    public AuthorDto Author { get; set; } = null!;
    public bool IsArchived { get; set; }
    public bool IsGlowing { get; set; }
}

File: mobile/design_courses/lib/data/course_card_dto.dart
class CourseCardDto {
  final int id;
  final String title;
  final String category;
  final String description;
  final String imageUrl;
  final int price;
  final bool isOwned;
  final bool isArchived;

  const CourseCardDto({
    required this.id,
    required this.title,
    required this.category,
    required this.description,
    required this.imageUrl,
    required this.price,
    required this.isOwned,
    required this.isArchived,
  });
}
